src/wp-includes/class-wp-meta-query.php

```diff
@@ -804,7 +833,7 @@ protected function find_compatible_table_alias( $clause, $parent_query ) {
 			if ( 'OR' === $parent_query['relation'] ) {
 				$compatible_compares = array( '=', 'IN', 'BETWEEN', 'LIKE', 'REGEXP', 'RLIKE', '>', '>=', '<', '<=' );
 
-				// Clauses joined by AND with "negative" operators share a join only if they also share a key.
+				// Clauses joined by AND need to have "negative" operators to share the join.
```

Comment:
This function isn't changing but with more specific joins my instinct is the rules for compatibility need to be tightened. Have you done any tests either way? 

I've looked over `tests/phpunit/tests/query/metaQuery.php` but can't see where it's tested.

Reply:
I've not done done any tests on those rules specifically, no. I've not come across any instances where the new query returns incorrect results, but I'm certainly not going to claim to have tried every possible combination! 

I think I flagged on trac that there may be some additional unit tests needed - sounds like this may be one such case.